src/test/java/io/jenkins/plugins/mcp/server/extensions/BuildLogExtensionTest.java

```diff
@@ -128,12 +134,51 @@ void testMcpToolCallWithIncorrectBuildNumber(JenkinsRule jenkins, JenkinsMcpClie
     }
 
     static Stream<Arguments> buildLogTestParameters() {
+
+        // all lines Started;[Pipeline] Start of Pipeline;[Pipeline] End of Pipeline;Finished: SUCCESS
+        // "Started;[Pipeline] Start of Pipeline;[Pipeline] End of Pipeline;Finished: SUCCESS"
         Stream<Arguments> baseArgs = Stream.of(
-                Arguments.of(100, null, 4, false),
-                Arguments.of(100, 1L, 3, false),
-                Arguments.of(3, null, 3, true),
-                Arguments.of(100, -1L, 1, false),
-                Arguments.of(3, -4L, 3, true));
+                Arguments.of(
+                        100,
+                        null,
+                        4,
+                        false,
+                        List.of(
+                                "Started",
+                                "[Pipeline] Start of Pipeline",
+                                "[Pipeline] End of Pipeline",
+                                "Finished: SUCCESS")),
+                Arguments.of(
+                        100,
+                        1L,
+                        3,
+                        false,
+                        List.of("[Pipeline] Start of Pipeline", "[Pipeline] End of Pipeline", "Finished: SUCCESS")),
+                Arguments.of(
+                        3,
+                        null,
+                        3,
+                        true,
+                        List.of("Started", "[Pipeline] Start of Pipeline", "[Pipeline] End of Pipeline")),
+                Arguments.of(
+                        100,
+                        -1L,
+                        4,
```

Comment:
@topikachu I was not sure to understand the previous logic of skip as negative value. 
My understand is `-1` with limit of 100 means start from lines and return 100 lines backward.

Reply:
No backward. 🤣
The test code only verifies that there is no out-of-bounds .

Reply:
:) 

with the changes we can get the last 100 lines by saying `skip=-1` with `limit=100`
something such `skip=-5` with `limit=100` will return 100 lines from 5 lines before the end.
Do you prefer the semantic to get the 100 last lines with the semantic ``skip=--100` with `limit=100`
TBH I don't mind much :) we only need to describe it correctly to be understandable by agent.
AND very important something we can implement without loading all the logs in memory.

Reply:
sounds good to me

Reply:
well this is a bit problematic. as this mean loading all the file in memory (as it is done currently and for very large build it's huge problem). I have tested this with some logs with 4M of lines, this was really bad.

Reply:
How about 
   `.setBufferSizeMax(bufferSizeMax)`
   when calling `ReversedLinesFileReader.builder()`

Reply:
this one is not the main issue :) 

here the summary 
I was afraid of the 2nd line, but that's doable (in a bit differently though) 
working on it

```
skip | limit | algo                                                                             |
-----|-------|----------------------------------------------------------------------------------|
10   |  100  |  ok  (logReader skip 10 then read 100)                                           |
100  | -10   |  ok  (cannot use logReader as need to keep 10 in a memory bucket read til 100)   |
-100 | 100   |  ok  (ReversedLinesFileReader keep 100 in memory read til 100 lines)             |
-100 | -100  |  ok  (ReversedLinesFileReader skip 100 then read 100)                            |
```

Reply:
100  | -10   |  ok  (cannot use logReader as need to keep 10 in a memory bucket read til 100)   |
=>
This means you need to skip 90 and read 10 more. It may not be clear or reasonable to handle the edge case, but let's make it work first. 

:)